File: solutions/python/isbn-verifier/1/isbn_verifier.py
def is_valid(isbn: str) -> bool:
    # Remove hyphens
    isbn = isbn.replace("-", "")
    
    # ISBN-10 must be exactly 10 chars
    if len(isbn) != 10:
        return False

    # First 9 chars must be digits
    if not isbn[:9].isdigit():
        return False

    # Last char can be a digit or 'X'
    if not (isbn[-1].isdigit() or isbn[-1] == "X"):
        return False

    total = 0
    for i, ch in enumerate(isbn):
        if ch == "X":
            value = 10
        else:
            value = int(ch)

        weight = 10 - i
        total += value * weight

    return total % 11 == 0

File: solutions/java/isbn-verifier/1/src/main/java/IsbnVerifier.java
class IsbnVerifier {

    boolean isValid(String stringToVerify) {

        String isbn = stringToVerify.replace("-", "");

        if (isbn.length() != 10) {
            return false;
        }

        int sum = 0;

        for (int i = 0; i < 10; i++) {
            char ch = isbn.charAt(i);
            int value;

            if (i == 9 && ch == 'X') {
                value = 10;
            } else if (Character.isDigit(ch)) {
                value = ch - '0';
            } else {
                return false;
            }

            sum += value * (10 - i);
        }

        return sum % 11 == 0;
    }
}